internal/controllers/machinedeployment/machinedeployment_controller.go

```diff
@@ -399,6 +410,70 @@ func (r *Reconciler) shouldAdopt(md *clusterv1.MachineDeployment) bool {
 	return !util.HasOwner(md.OwnerReferences, clusterv1.GroupVersion.String(), []string{"Cluster"})
 }
 
+func (r *Reconciler) adjustManagedFields(ctx context.Context, obj client.Object) error {
```

Comment:
nit: adjust sounds very generic, WDYT about calling this func cleanupLegacyManagedFields or something similar?

Let's also add some high-level godoc comment here what we're doing

Reply:
Same in MS controller

Reply:
Move the function to a separate package. Will add unit tests.